src/components/map/LayersService.js

```diff
@@ -799,44 +806,43 @@ goog.require('ga_urlutils_service');
                   });
                 });
           } else if (config.type === 'vectortile') {
-
-            if (config.sourceType === 'raster') {
-              olLayer = new ol.layer.Tile();
-
-            } else {
-              olLayer = new ol.layer.VectorTile({
-                declutter: true,
-                style: new ol.style.Style(),
-                source: new ol.source.VectorTile({
-                  format: new ol.format.MVT(),
-                  maxZoom: config.maxZoom,
-                  url: config.url
-                })
-              });
-            }
+            olLayer = new ol.layer.VectorTile({
+              declutter: true,
+              style: new ol.style.Style(),
+              source: new ol.source.VectorTile({
+                format: new ol.format.MVT(),
+                maxZoom: config.maxZoom,
+                url: config.url
+              })
+            });
+            gaDefinePropertiesForLayer(olLayer);
             olLayer.setOpacity(config.opacity || 1);
-
-            if (config.sourceId && styleUrl) {
-              olLayer.sourceId = config.sourceId;
+            var applyTilesetJson = function(olLayer, glStyle) {
+              // Load informations from tileset.json file of a source
+              var sourceConfig = glStyle.sources[olLayer.sourceId];
+              sourceConfig.minZoom = config.minZoom;
+              sourceConfig.maxZoom = config.maxZoom;
+              if (sourceConfig) {
+                gaMapUtils.applyGlSourceToOlLayer(olLayer, sourceConfig);
+              }
+            };
+            var sourceId = config.sourceId;
+            if (sourceId) { olLayer.sourceId = sourceId; }
+            if (glStyle) {
+              gaMapUtils.applyGlStyleToOlLayer(olLayer, glStyle);
+              applyTilesetJson(olLayer, glStyle);
+            } else if (styleUrl) {
```

Comment:
Here I propose a small refactoring , it will avoid to create a function before, help to understand better the code:
``` 
var p =  (glStyle) ? $q.when(glStyle) : gaGlStyle.get(styleUrl);
p.then(function(glStyle) {
￼      if (!glStyle) {		￼  
￼          return;
￼      }	
```

Reply:
yep

Reply:
yeah but styleUrl must be defined as well if we want to perform a request that is not always the case

Reply:
I can do that

```javascript
            if (glStyle || styleUrl) {
              p =  (glStyle) ? $q.when(glStyle) : gaGlStyle.get(styleUrl);
              p.then(function(glStyle) {
                createSubLayers(olLayer, glStyle);
              });
            } else {
              createSubLayers(olLayer);
            }
```
if you prefer